File: src/test/java/warmke/mike/leet/p42/SolutionTest.java
package warmke.mike.leet.p42;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import warmke.mike.leet.two.pointers.p42.Solution;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SolutionTest {

    Solution solution;

    @BeforeEach
    public void setup() {
        solution = new Solution();
    }

    @Test
    public void test() {
        assertEquals(6, solution.trap(new int[]{0,1,0,2,1,0,1,3,2,1,2,1}));
        assertEquals(9 , solution.trap(new int[]{4,2,0,3,2,5}));
    }
}

File: src/test/java/warmke/mike/leet/p217/SolutionTest.kt
package warmke.mike.leet.p217

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import warmke.mike.leet.arrays.p217.Solution

internal class SolutionTest {
    val s = Solution()

    @Test
    fun test() {
        assertEquals(true ,s.containsDuplicate(intArrayOf(1,2,3,1)))
        assertEquals(false ,s.containsDuplicate(intArrayOf(1,2,3,4)))
    }

}
